### PR TITLE
Dissalow wrong annotations in assignments

### LINKED ISSUE #925: Type annotation check doesn't work for type aliases
# Bug report

## What's wrong
Type annotation check doesn't work if type aliases are used. For example, this code
```
type_alias = Literal[None]

def my_func(arg: type_alias):
    pass
```
passes flake8 checks without raising `WPS701` error, while this code
```
def my_func(arg: Literal[None]):
    pass
```
makes flake8 complain: "WPS701 Found useless `Literal[None]` typing annotation".
## How is that should be
Type annotation errors should be raised whether type aliases are used or not.

## System information
wemake-python-styleguide: 0.12.5

### PR DESCRIPTION
**Pull request by AlwxSin**
# I have made things!

## Checklist

- [X] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [X] I have created at least one test case for the changes I have made
- [X] I have updated the documentation for the changes I have made
- [X] I have added my changes to the `CHANGELOG.md`

## Related issues
closes #925

As a side effect, fixed several wrong multiline annotations.

### CODE REVIEW

File: wemake_python_styleguide/violations/annotations.py [hunk 1]
```diff
@@ -62,7 +64,7 @@ def func(empty: Literal[None]):
     """
 
     code = 700
-    error_template = 'Found useless `Literal[None]` typing annotation'
```
[the remark this answers is not in the document]
Comment (sobolevn) [reply wemake_python_styleguide/violations/annotations.py]: You can remove the word `typing` 🙂

File: wemake_python_styleguide/violations/annotations.py [hunk 2]
```diff
@@ -87,6 +89,7 @@ class NestedAnnotationsViolation(ASTViolation):
         # Wrong:
         Literal[Literal[Literal[1, 2, 3], "foo"], 5, None]
         Union[Union[int, str], float]
+        NONE_ALIAS = Union[Union[int, str]
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/annotations.py]: I guess the existing example is good enough. It is about types, not their usage.

File: wemake_python_styleguide/violations/annotations.py [hunk 3]
```diff
@@ -115,6 +118,7 @@ class UnionNestedInOptionalViolation(ASTViolation):
 
         # Wrong:
         Optional[Union[int, str]]
+        NONE_ALIAS = Optional[Union[int, str]]
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/annotations.py]: The same here.

File: wemake_python_styleguide/visitors/ast/annotations.py [hunk 4]
```diff
@@ -31,6 +31,11 @@ class _GenericAnnotationVisitor(BaseNodeVisitor):
         'typing_extensions.',
     ))
 
+    def visit_Assign(self, node: ast.Assign) -> None:
+        """Checks assignment patterns."""
+        self._check_annotation(node)
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/annotations.py]: Does it cover `x = y = Literal[None]` and `x, y = Literal[None], Literal[None]`?
We need to test it.

Comment (AlwxSin) [reply wemake_python_styleguide/visitors/ast/annotations.py]: It should. I'll add more tests.

File: wemake_python_styleguide/violations/annotations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlwxSin) [reply wemake_python_styleguide/violations/annotations.py]: After a night I think that I need to exclude strings from checking completely. Or this code will be invalid too
```python
def get_example_of_wrong_annotation:
    wrong1 = 'Literal[None]'
    wrong2 = Optional[Union[str, float]]
    return (wrong1, wrong2)
```

File: wemake_python_styleguide/visitors/ast/annotations.py (hunk 4 above)
Comment (AlwxSin) [reply wemake_python_styleguide/visitors/ast/annotations.py]: Added more tests.

File: wemake_python_styleguide/violations/annotations.py (hunk 3 above)
Comment (AlwxSin) [reply wemake_python_styleguide/violations/annotations.py]: Removed.

File: wemake_python_styleguide/violations/annotations.py (hunk 2 above)
Comment (AlwxSin) [reply wemake_python_styleguide/violations/annotations.py]: Removed.

File: wemake_python_styleguide/violations/annotations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlwxSin) [reply wemake_python_styleguide/violations/annotations.py]: So, now I don't check if it is string in assignment.

File: tests/test_visitors/test_ast/test_annotations/test_annotation_metadata/conftest.py [hunk 5]
```diff
@@ -7,6 +7,22 @@ def function(arg: {0}):
...
+function_arg_alias_template = """
+Type_Alias = {0}
+
+def function(arg: Type_Alias):
+    ...
+"""
+
+function_arg_multiple_alias_template = """
+a, Another_Alias = {0}, str
+b = c = {0}
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_annotations/test_annotation_metadata/conftest.py]: Sorry, this does not look correct to me. 

We need to make three separate new templates for this case:
1. `a = {0}`
2. `b = c = {0}`
3. `d, e = {0}, {0}`

And nothing else, function is redundant here. 

I am also not sure about how `d, e = {0}, {0}` should behave. I guess, that it should raise two violations. However, it is not critical and one is good enough.
See which implementation is easier.

Thanks a lot for your time and effort! 👍

Comment (AlwxSin) [reply tests/test_visitors/test_ast/test_annotations/test_annotation_metadata/conftest.py]: Will do.

But with `d, e = Literal['a'], Literal['b']` I have a false positive `NestedAnnotationsViolation`. Not sure how to defeat it.

Comment (AlwxSin) [reply tests/test_visitors/test_ast/test_annotations/test_annotation_metadata/conftest.py]: Well, regex will always help 😁
Fixed.

**Comment by sobolevn** [comment]
Thanks a lot!

**Pull request merged**